--- vendor/phalcon/devtools/api/Phalcon/DI/Injectable.php.orig	2015-04-10 01:45:05.912108123 +0200
+++ vendor/phalcon/devtools/api/Phalcon/DI/Injectable.php	2015-04-10 02:04:21.371645795 +0200
@@ -1,5 +1,30 @@
 <?php 
 
+namespace OpenExam\Library\DI {
+
+    class Logger
+    {
+
+        /**
+         * The system log.
+         * @var \Phalcon\Logger\AdapterInterface 
+         */
+        public $system;
+        /**
+         * The authentication log.
+         * @var \Phalcon\Logger\AdapterInterface
+         */
+        public $auth;
+        /**
+         * The debug log.
+         * @var \Phalcon\Logger\AdapterInterface
+         */
+        public $debug;
+
+    }
+
+}
+
 namespace Phalcon\DI {
 
     /**
@@ -180,5 +205,82 @@
         {
         }
 
+        // 
+        // OpenExam specific services:
+        // 
+
+        /**
+         * The access control (ACL) service.
+         * @var \OpenExam\Library\Security\Acl
+         */
+        public $acl;
+        /**
+         * The user authentication service.
+         * @var \OpenExam\Library\Security\Authentication
+         */
+        public $auth;
+        /**
+         * The multi-level general purpose caching service.
+         * @var \OpenExam\Library\Core\Cache
+         */
+        public $cache;
+        /**
+         * The capabilities service.
+         * @var \OpenExam\Library\Security\Capabilities
+         */
+        public $capabilities;
+        /**
+         * The catalog (directory information) service.
+         * @var \OpenExam\Library\Catalog\DirectoryManager
+         */
+        public $catalog;
+        /**
+         * Database connection object (for read only).
+         * @var \Phalcon\Db\Adapter\Pdo 
+         */
+        public $dbread;
+        /**
+         * Database connection object (for read/write).
+         * @var \Phalcon\Db\Adapter\Pdo 
+         */
+        public $dbwrite;
+        /**
+         * The locale service.
+         * @var \OpenExam\Library\Globalization\Locale\Locale
+         */
+        public $locale;
+        /**
+         * The location and access information service.
+         * @var \OpenExam\Library\Core\Location
+         */
+        public $location;
+        /**
+         * The logging service.
+         * 
+         * <code>
+         * $this->logger->system->error($message);                      // Write error to system log
+         * $this->logger->auth->info($message);                         // Log authentication event
+         * $this->logger->auth->log($message, LOGGER::INFO);            // Log authentication event
+         * $this->logger->debug->log(__METHOD__ . ': ' . $message);     // Using the debug log
+         * </code>
+         * @var \OpenExam\Library\DI\Logger
+         */
+        public $logger;
+        /**
+         * The render service.
+         * @var \OpenExam\Library\Render\RenderService
+         */
+        public $render;
+        /**
+         * The translation service.
+         * @var \OpenExam\Library\Globalization\Translate\Translate
+         */
+        public $tr;
+        /**
+         * The authenticated user.
+         * @var \OpenExam\Library\Security\User 
+         */
+        public $user;
+        
     }
 }
